src/ebonite/api/instances.py

```diff
@@ -38,52 +57,142 @@ def get_instances() -> Tuple[Response, int]:
                                         f'Image with id {instance.image_id} does not exist'}), 404
         try:
-            instance = ebonite.run_instance(name=instance.name, image=image, environment=env)
-            return jsonify(pj.dumps(instance)), 201
+            instance = ebonite.create_instance(name=instance.name, image=image,
+                                               environment=env, run=run, runner_kwargs=runner_kwargs, **instance_kwargs)
+            return jsonify(pj.serialize(instance)), 201
         except ExistingInstanceError:
             return jsonify({'errormsg': f'Instance with name {instance.name} '
                                         f'and image {image.name} already exists'}), 400
+        except (ValueError, TypeError) as e:
+            return jsonify({'errormsg': str(e)}), 404
 
-    @blueprint.route('/<int:id>', methods=['PATCH'])
-    def update_instance(id: int) -> Tuple[Response, int]:
-        body = request.get_json(force=True)
-        body['id'] = id
-        instance = UpdateInstanceBiody.from_data(body)
-        try:
-            ebonite.meta_repo.update_instance(instance)
-            return jsonify({}), 204
-        except NonExistingInstanceError:
-            return jsonify({'errormsg': f'Instance with id {id} does not exist'}), 404
+    # @blueprint.route('/<int:id>', methods=['PATCH'])
+    # def update_instance(id: int) -> Tuple[Response, int]:
+    #     # TODO: Same as with image - to update we need to recreate instance?
+    #     body = request.get_json(force=True)
+    #     body['id'] = id
+    #     instance = UpdateInstanceBiody.from_data(body)
+    #     try:
+    #         ebonite.meta_repo.update_instance(instance)
+    #         return jsonify({}), 204
+    #     except NonExistingInstanceError:
+    #         return jsonify({'errormsg': f'Instance with id {id} does not exist'}), 404
 
     @blueprint.route('/<int:id>', methods=['DELETE'])
     def delete_instance(id: int) -> Tuple[Response, int]:
+        """
+        Deletes and, optionally, stops instance
+        ---
+        parameters:
+          - name: id
+            in: path
+            type: integer
+            required: true
+          - name: meta_only
+            in: query
+            type: integer
+            required: false
+            default: 0
+        responses:
+          204:
+            description: Image succesfully deleted
+          404:
+            description: Image with given id does not exist
+        """
+        meta_only = False if not request.args.get('meta_only') else bool(int(request.args.get('meta_only')))
         instance = ebonite.meta_repo.get_instance_by_id(id)
         if instance is None:
             return jsonify({'errormsg': f'Instance with id {id} does not exist'}), 404
-        ebonite.stop_instance(instance)
+        ebonite.delete_instance(instance, meta_only=meta_only)
         return jsonify({}), 204
```

Comment:
why return empty object in body? either smth like {ok:True} or just empty string will be better i think

Reply:
204 response body is always empty

Reply:
can you just return None, 204 or '', 204?